test/exe/main_common_test.cc

```diff
@@ -25,14 +25,25 @@
 
```

Comment:
Can't comment on line 22. The ifdef for [ENVOY_HOT_RESTART}(https://github.com/envoyproxy/envoy/pull/2679/files#diff-a1cde380f86310682f6d87cfbbd0c5f5R22) isn't needed anymore!

Reply:
See @mattklein123's comment in https://github.com/envoyproxy/envoy/issues/2029

"@tonya11en I would probably leave the compile flag as some people want the ability to just compile out the functionality entirely. I think doing a CLI flag alongside would be pretty straightforward."

Reply:
I'm fine letting Matt's opinion stand, but we're not actually compiling out the functionality, we're compiling out the ability to configure the functionality.  I think it's cleaner to just have people turn it off but I'll totally merge this as a huge improvement in readability as it stands :-)

Reply:
I thought in the old code we actually used the bazel switch to not compile the full hot restart implementation and only compile the NOP implementation. This might not be the case. I don't feel strongly about this either way. If people think it's better to get rid of the compile option we can do that. I suspect almost no one is using that option and as long as we have a clear release note (please check) it's probably OK. If we do that please remember to cleanup the bazel README docs about the option also.

Reply:
Ooh, indeed, I was grepping source/... only and missed it in bazel/envoy_build_system.bzl

Reply:
I was commenting on the lines 22-24 not being required in this file because the header "#include "server/hot_restart_impl.h"" isn't used in this file.

Reply:
Good catch Carmen!  Done.

Reply:
RE removing the ifdef -- if no one thinks we need this ifdef anymore I'll remove it in a future PR.  I think before we do that we just need to first make sure everyone who doesn't want hot-restart sets the new flag (after this is checked in).